Source/Csla/Core/ContextDictionary.cs

```diff
@@ -72,6 +73,101 @@ void IMobileObject.SetChildren(SerializationInfo info, MobileFormatter formatter
+    /// <exception cref="System.ArgumentException">An element with the same key already exists in the System.Collections.IDictionary object.</exception>
+    /// <exception cref="System.NotSupportedException">The System.Collections.IDictionary is read-only. -or- The System.Collections.IDictionary has a fixed size.</exception>
+    public void Add(object key, object value)
+    {
+      ((IDictionary)this).Add(key, value);
+    }
+
+    /// <summary>
+    /// Removes the element with the specified key from the System.Collections.IDictionary
+    /// object.
+    /// </summary>
+    /// <param name="key">The key of the element to remove.</param>
+    /// <exception cref="System.ArgumentNullException">key is null.</exception>
+    /// <exception cref="System.NotSupportedException">The System.Collections.IDictionary object is read-only. -or- The System.Collections.IDictionary has a fixed size.</exception>
+    public void Remove(object key)
+    {
+      ((IDictionary)this).Remove(key);
+    }
+
+    #endregion
+
+    #region ICollection Members
+
+    /// <summary>
+    /// Gets an object that can be used to synchronize access to the System.Collections.ICollection.
+    /// </summary>
+    /// <returns>An object that can be used to synchronize access to the System.Collections.ICollection.</returns>
+    public object SyncRoot
+    {
+      get => ((IDictionary)this).SyncRoot;
+    }
+
+    /// <summary>
+    /// Gets a value indicating whether access to the System.Collections.ICollection
+    /// is synchronized (thread safe).
+    /// </summary>
+    ///<returns>true if access to the System.Collections.ICollection is synchronized (thread safe); otherwise, false.</returns>
+    public bool IsSynchronized
+    {
+      get => ((ICollection)this).IsSynchronized;
+    }
+
+    /// <summary>
+    /// Copies the elements of the System.Collections.ICollection to an System.Array,
+    /// starting at a particular System.Array index.
+    /// </summary>
+    /// <param name="array">
+    /// The one-dimensional System.Array that is the destination of the elements copied
+    /// from System.Collections.ICollection. The System.Array must have zero-based indexing.
+    /// </param>
+    /// <param name="index">The zero-based index in array at which copying begins.</param>
+    /// <exception cref="System.ArgumentNullException">array is null.</exception>
+    /// <exception cref="System.ArgumentOutOfRangeException">index is less than zero.</exception>
+    /// <exception cref="System.ArgumentException">
+    /// array is multidimensional. -or- The number of elements in the source System.Collections.ICollection
+    /// is greater than the available space from index to the end of the destination
+    /// array. -or- The type of the source System.Collections.ICollection cannot be cast
+    /// automatically to the type of the destination array.
+    /// </exception>
+    public void CopyTo(Array array, int index) 
```

Comment:
I think some of these documentions can just be inherited. Like this one. Instead you could use `/// <inheritdoc cref="System.Collections.ICollection.CopyTo(Array, int)"/>` which uses the documentation of the provided cref.

Reply:
Oh, that's nice - I didn't know that was possible.

Reply:
I didn't know that was possible either. That would have saved some time! :)

Reply:
And yes, it does indeed work...